src/components/Header/Header.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useEffect, useRef, useCallback } from 'react';
+import classNames from 'classnames';
+import { TodosContext } from '../../TodosContext';
+
+export const Header: React.FC = () => {
+  const {
+    todos,
+    title,
```

Comment:
do you need to save this field in context?  I suggest to you it will be better for the performance and readability  if you create the title state here, and pass this title as an argument to addTodo